src/cmd/update.rs

```diff
@@ -0,0 +1,73 @@
+use std::env;
+use std::fs;
+use std::fs::Permissions;
+use std::os::unix::prelude::*;
+use std::path::Path;
+
+use crate::tempfile::Tempfile;
+use anyhow::Result;
+use serde::Deserialize;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+const RELEASES_LIST: &str = "https://api.github.com/repos/sbstp/kubie/contents/releases/linux?ref=master";
+const RELEASE_BASE_URL: &str = "https://github.com/sbstp/kubie/raw/master/releases/linux/amd64";
+const FILENAME: &str = "kubie";
+
+#[derive(Debug, Deserialize)]
+struct TreeUrl {
+    git_url: String,
+}
+
+#[derive(Debug, Deserialize)]
+struct Tree {
+    tree: Vec<KubieVersion>,
+}
+
+#[derive(Debug, Deserialize)]
+struct KubieVersion {
+    path: String,
+}
+
+pub fn update() -> Result<()> {
+    let latest_version = get_latest_version()?;
+    if latest_version.eq(&format!("v{}", VERSION)) {
+        println!("Kubie is up-to-date : v{}", VERSION);
+    } else {
+        println!(
+            "A new version of Kubie is available ({}), the new version will be automatically installed...",
+            latest_version
+        );
+        let resp = attohttpc::get(format!("{}/{}/{}", RELEASE_BASE_URL, latest_version, FILENAME)).send()?;
+        if resp.is_success() {
+            let tmp_file = Tempfile::new("/tmp", "kubie", "")?;
+            resp.write_to(&*tmp_file)?;
+            let old_file = env::current_exe().expect("could not get own binary path");
+            let res = replace_file(&old_file, tmp_file.path());
+            match res {
+                Ok(_) => {
+                    println!(
+                        "Kubie has been updated successfully. Enjoy :) ({})",
+                        old_file.to_string_lossy()
```

Comment:
You can use `Path::display` to print a path without having to allocate. 

https://doc.rust-lang.org/std/path/struct.Display.html

Reply:
Done :) Thanks